spec/integration/slices_spec.rb

```diff
@@ -81,6 +81,188 @@ class Slice < Hanami::Slice
     end
   end
 
+  it "Loading a slice with its own defined slice class" do
+    with_tmp_directory(Dir.mktmpdir) do
+      write "config/app.rb", <<~RUBY
+        require "hanami"
+
+        module TestApp
+          class App < Hanami::App
+          end
+        end
+      RUBY
+
+      write "slices/main/config/slice.rb", <<~RUBY
+        module Main
+          class Slice < Hanami::Slice
+            config.actions.format :json
+          end
+        end
+      RUBY
+
+      require "hanami/prepare"
+
+      expect(Hanami.app.slices[:main]).to be Main::Slice
+      expect(Hanami.app.slices[:main].config.actions.format).to eq([:json])
+    end
+  end
+
+  it "Loading a slice with its own defined slice class prefers the app's defined slice class" do
+    with_tmp_directory(Dir.mktmpdir) do
+      write "config/app.rb", <<~RUBY
+        require "hanami"
+
+        module TestApp
+          class App < Hanami::App
+          end
+        end
+      RUBY
+
+      write "config/slices/main.rb", <<~RUBY
+        require "hanami"
+
+        module Main
+          class Slice < Hanami::Slice
+            config.actions.format :html
+          end
+        end
+      RUBY
+
+      write "slices/main/config/slice.rb", <<~RUBY
+        module Main
+          class Slice < Hanami::Slice
+            config.actions.format :json
+          end
+        end
+      RUBY
```

Comment:
My understanding of the implementation is that, in this scenario, this file here should never get loaded at all. Is that right?

To make that even clearer for a test like this, could we actually just have the file look something like this?

```ruby
write "slices/main/config/slice.rb", <<~RUBY
  raise "I should never be loaded"
RUBY
```

I tested this locally and everything still passes. What do you think?

If we did make this change, we should probably change the one remaining slice class to use  `.format :json` just for overall consistency.

Reply:
Ah yeah, good call